include/git2/branch.h

```diff
@@ -246,6 +246,18 @@ GIT_EXTERN(int) git_branch_is_head(
 	const git_reference *branch);
 
 /**
+ * Determine if the current branch is checked out in any linked
+ * repository.
+ *
+ * @param branch Reference to the branch.
+ *
+ * @return 1 if branch is checked out, 0 if it isn't,
+ * error code otherwise.
+ */
+GIT_EXTERN(int) git_branch_is_checked_out(
```

Comment:
Well, I agree this interface should be modified. Currently, we simply return `false` even if we were not able to open a worktree, which might not be the right thing to do here. So three possibilities here:

1. Add a flag to `git_branch_is_checked_out`, which indicates whether to honor working trees at all and, if they should be honored, what to do if a working tree is not present.
2. Have some kind of return code here that indicates that some working trees could not be inspected.
3. The ideal thing would be to modify `git_branch_is_head` here to accept the new flag, saying how it should actually behave. While at it and breaking the API, I think it is a bit awkward that the function only accepts a branch without any repository - while the repository is retrievable via the branch itself, I find it a bit restricting.

So I'd not really want to coalesce these two functions together without breaking the API here, as it might not always be the case that a user wants to know if the branch is checked out in _any_ linked repository. Opinions?